src/lib/handlers/db.js

```diff
@@ -0,0 +1,28 @@
+import { db } from '$lib/firebase/firebase';
+import { collection, doc, getDoc, getDocs, setDoc } from 'firebase/firestore';
+
+export const getUserTricks = async (user) => {
+	if (!user) return [];
+	const userRef = doc(db, 'users', user.uid);
+	const userSnap = await getDoc(userRef);
+	if (!userSnap.exists()) {
+		console.log('No data for this user found in the collection.');
+		const newUser = doc(db, 'users', user.uid);
```

Comment:
not a big deal but we can extract the creating user document function to make it more readable

Reply:
Great point! created a separate function with the last commit